scripts/validation_utils.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for schema and command line validation"""
+
+import argparse
+import re
+
+
+# For easier development, we allow redefining builtins like
+# --substitutions=PROJECT_ID=foo even though gcloud doesn't.
+KEY_VALUE_REGEX = re.compile(r'^([A-Z_][A-Z0-9_]*)=(.*)$')
+
+
+def get_field_value(container, field_name, field_type):
```

Comment:
Don't be distressed, I'm not trying to say "never type check"- I'm just trying to understand why you're doing it. Python is still a strongly typed language so if you try to make something quack that isn't a duck, the user will still get an error. All this "manual" type checking does is more or less present a nicer (and earlier) error message. Checking a type and validating values are both fine, I just want to understand the context. If your purpose here is to provide a helpful error to the user (rather than a generic attributeerror) or similar, then I'm cool with it.

For what its worth, I question classes the same way. Sometimes they're useful, but I sometimes want to understand the motivation. A question isn't an attack on your code or your motivations, it's just me trying to understand why so that I can better review. :)